automotive/odcantools/libodcantools/src/CANMessage.cpp

```diff
@@ -0,0 +1,365 @@
+#include <limits>
+#include <cmath>
+#include <iostream>
+#include <algorithm> // for std::rotate and std::transform
+
+#include "CANMessage.h"
+
+// the implemented logic is based on the resource: www.race-technology.com/wiki/index.php/CANInterface/ByteOrdering 
+// implemented cases of interest are "Intel Standard" and "Motorola Forward MSB"
+
+namespace automotive {
+    namespace odcantools {
+        CANMessage::CANMessage(const GenericCANMessage& gcm) :
+            GenericCANMessage(gcm),
+            m_payload(),
```

Comment:
We should preallocate 8 bytes in our vectors to avoid vector copying back and forth at runtime.

Please check if a preallocated vector does not collide with your logic (i.e., when you check for payload.empty() for instance)